linkerd/app/outbound/src/http/proxy_connection_close.rs

```diff
@@ -83,23 +83,43 @@ impl<B, F: TryFuture<Ok = http::Response<B>>> Future for ResponseFuture<F> {
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
         let this = self.project();
-        let rsp = futures::ready!(this.inner.try_poll(cx))?;
+        let mut rsp = futures::ready!(this.inner.try_poll(cx))?;
 
-        if let Some(proxy_conn) = rsp.headers().get(L5D_PROXY_CONNECTION) {
+        // Clear the header...
+        if let Some(proxy_conn) = rsp.headers_mut().remove(L5D_PROXY_CONNECTION) {
             if proxy_conn == "close" {
-                if let Some(error) = rsp
-                    .headers()
-                    .get(L5D_PROXY_ERROR)
-                    .and_then(|v| v.to_str().ok())
-                {
-                    tracing::info!(%error, "Closing application connection for remote proxy");
-                } else {
-                    tracing::info!("Closing application connection for remote proxy");
+                match rsp.extensions().get::<tls::ConditionalClientTls>() {
+                    Some(tls::ConditionalClientTls::Some(_)) => {
+                        if let Some(error) = rsp
+                            .headers()
+                            .get(L5D_PROXY_ERROR)
+                            .and_then(|v| v.to_str().ok())
+                        {
+                            tracing::info!(%error, "Closing application connection for remote proxy");
+                        } else {
+                            tracing::info!("Closing application connection for remote proxy");
+                        }
```

Comment:
...whew okay i should really merge the `impl Value for Option<T: Value>` PR in `tracing`...